src/lib/controllers/upload-controller.js

```diff
@@ -76,4 +77,27 @@ export default class UploadController {
       targetDocSources
     }
   }
+
+  static plainSourceUploadFromFileSingle ({ fileData, textId, textType, tokenization }) {
+    if (fileData.length === 0 || fileData.search(/\r\n|\r|\n/) === -1) {
+      console.error(L10nSingleton.getMsgS('UPLOAD_CONTROLLER_ERROR_WRONG_FORMAT'))
+      NotificationSingleton.addNotification({
+        text: L10nSingleton.getMsgS('UPLOAD_CONTROLLER_ERROR_WRONG_FORMAT'),
+        type: NotificationSingleton.types.ERROR
+      })
+      return
+    }
+    const fileDataArr = fileData.split(/\r\n|\r|\n/)
+
+    if (!Array.isArray(fileDataArr) || fileDataArr.length < 4) {

```

Comment:
I don't understand the restriction for at least 4 lines. I think it should be possible to do an alignment of a single line of text.

Reply:
As for text - it could contatin as much lines as it is needed.
Line breaks rare  replaced with another character.

@balmas ,I didn't spend much time to download/upload formats - and made it only like a first variant.
But if you don't like it - I could refactor the file format:
The first line would contain all text properties: lang, direction, sourceType, the amount of text lines
And all others would contain all the text with its origin line breaks.

What do you think?

Reply:
I decided to update the download/upload format
now it looks like this:
```
HEADER: 1	ltr	lat	text
Test 1
Test 1
HEADER: 2	ltr	ara	text
Test 2
Test 2
HEADER: 3	ltr	eng	text
Test 3
Test 3
```